rust-runtime/aws-smithy-http/src/byte_stream.rs

```diff
@@ -445,6 +445,12 @@ impl From<hyper::Body> for ByteStream {
     }
 }
 
+impl From<lambda_http::Body> for ByteStream {
+    fn from(input: lambda_http::Body) -> Self {
+        ByteStream::new(SdkBody::from(input))
+    }
+}
```

Comment:
this shouldn't be needed since `lambda_http::Body` can be converted into `BoxBody` right?

Reply:
It's to support the event streaming case. We have an `B: Into<aws_smithy_http::byte_stream::ByteStream>` bound on server operation handlers that do event streams, because eventually there's an

```
crate::http_serde::deser_payload_capture_pokemon_operation_capture_pokemon_operation_input_events(
    &mut body.into().into_inner(),
)?,
```

We convert the request body type (remember in the server we support all body types so long as it implements `HttpBody`) into `SdkBody` before sending it to the deserializer, which just works with `SdkBody`. All of this would go away if we tackled https://github.com/awslabs/smithy-rs/issues/1125 and made `ByteStream` work with all body types.